### ISSUE TITLE
Spacing Issues for the Beta Evaluation Button

### ISSUE THREAD

**Issue by Patrick-123** [unicorn-fail/dreditor#253]
It looks like the Beta Evaluation Button does not produce the same spacing as the template. This bunches the first paragraph in one block of code that isn't intended based on the template. 

What the current button produces:
![beta evaluation bug initial output](https://cloud.githubusercontent.com/assets/6172311/7442974/69f2be70-f0de-11e4-8921-56e4ed594f86.png)

What the template looks like:
![beta evaluation template](https://cloud.githubusercontent.com/assets/6172311/7442976/7d0bc9b6-f0de-11e4-8171-524796a77abb.png)

What the button should produce:
![how it should look when you output the button](https://cloud.githubusercontent.com/assets/6172311/7442982/970547f2-f0de-11e4-8847-4f0c9b2a642c.png)

I took a look at the code. It looks like we're using a regular expression to look for the closing comment [-->] and then putting /n/n after every instance of that. So it looks like for now we could just add a comment for every instance we want spacing on the Drupal template [https://www.drupal.org/node/2373483] so it will look legit before all the Sprints at DrupalCon LA but adjusting the template to fit the code and not the other way around doesn't seem to be the right way to go about it.

**Comment by Patrick-123** [comment]
YesCT was gracious enough to edit the Remaining Tasks template to test the add a comment theory. I do not have permissions so it wouldn't be possible without her aid. 

The Insert Tasks button has the same spacing issues as the Beta Evaluation button.

https://www.drupal.org/node/2272209/revisions/view/7944379/8421955

Based on the results, that theory holds. 
![remaining tasks template change](https://cloud.githubusercontent.com/assets/6172311/7443006/2dd7820c-f0e0-11e4-9b8a-206f4bce0afd.png)

**Comment by YesCT** [comment]
Thanks for opening this issue.

**Comment by star-szr** [comment]
Yes, this has always been a weak point. I'm looking at this now to see if we can fix this in a better way.

**Comment by markhalliwell** [comment]
FWIW, the current code (https://github.com/dreditor/dreditor/blob/1.x/src/js/plugins/issue.summary.template.js#L111) that does this is scraping the page manually via an AJAX and then using `$.text()` to grab the contents (which doesn't really respect whitespace): http://stackoverflow.com/questions/8177394/how-to-get-whitespace-using-jquerys-html-or-text

I would suggest instead we attempt to use [d.o's native API](https://www.drupal.org/api) (#159) that is in place now to retrieve this:
https://www.drupal.org/api-d7/node/2373483.json

edit: The only "downsides" I can see with this is that the body field should really only be the template itself (no preface text) and the HTML shouldn't be encoded already (by the code filter?).

**Comment by star-szr** [comment]
That sounds sensible, in the meantime using the native `innerText` property preserves newlines.

**Comment by markhalliwell** [comment]
Possibly, I cannot remember if that is a consistent property to use or not (http://perfectionkills.com/the-poor-misunderstood-innerText/). It may be necessary to use `el.textContent || el.innerText`, but not entirely sure.

**Comment by star-szr** [comment]
Dang. Well textContent doesn't preserve the newlines :/

**Comment by markhalliwell** [comment]
After talking with @Cottser about this in IRC, there's quite a few issues regarding any sort of "fix" for this:
1. As mentioned above, there are inconsistencies with how `innerText` works with newlines cross-browser.
2. We had discussed several methods on how we should "store" these templates but ultimately they're just different methods of "hacks" and each and every one will have their own pros/cons. There isn't a great "solution" for this at the moment.
3. The current method that this template is stored (a node on d.o) is abstract, at best, and can easily change at worst. From a code development standpoint, this sort of arbitrary "template" is volatile and is ultimately just a hack that Dreditor uses to inject into a text area. We want to move away from this, it's not safe and is actually preventing Dreditor from being accepted into https://addons.mozilla.org.
4. We have decided that this is really a d.o issue and any sort of "fix" should instead be discussed at https://www.drupal.org/node/1393226 so we can actually remove this "feature" from Dreditor.
